extensions/test-src/org/pentaho/platform/web/http/api/resources/UserSettingsResourceTest.java

```diff
@@ -46,8 +56,29 @@ public void teardown() {
   public void testSetUserSetting() throws Exception {
     IUserSettingService userSettingService = mock( IUserSettingService.class );
     PentahoSystem.registerObject( userSettingService );
-    Response response = userSettingsResource.setUserSetting( "favorites", "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+    Response response = userSettingsResource.setUserSetting( USER_SETTING_NAME, "[{\"fullPath\":\"/public/Steel Wheels/Top Customers (report).prpt\", \"title\":\"<script>if" + "(true&&'str'!='str2')alert(1)</script>\", \"lastUse\":0}]" );
+
+    verify( userSettingService ).setUserSetting( USER_SETTING_NAME,      USER_SETTING_VALUE );
```

Comment:
Not sure what's tested here either. You called a mock with an unmocked method, then verifying that you indeed called it one line earlier?